src/deltacode/__init__.py

```diff
@@ -48,16 +49,38 @@ class DeltaCode(object):
     the form of File objects) contained in those scans.
     """
     def __init__(self, new_path, old_path, options):
-        self.new = Scan(new_path)
-        self.old = Scan(old_path)
+        self.codebase1 = None
+        self.codebase2 = None
+        self.new_files_fingerprint = dict() # map of { {new_file1:fingerprint},{new_file2:fingerprint},...} it will be needed when we need the fingerprints
+        self.old_files_fingerprint = dict() # map of { {old_file1:fingerprint},{old_file2:fingerprint},...}
+        self.new_files_original_path = dict() #this keeps a map of the path of file with respect to original path 
+
+        """ 
+            in self.new_files_fingerprint we keep the key as files original path and the value as its fingerprint value
+        """
+        self.old_files_original_path = dict()
+
+        """
+            in self.old_files_original_path we keep key as the path which is retured by align scan and value as
+            the original path.
+            This original path needs to be cached as we require it for the report.
+        """
         self.options = options
         self.deltas = []
         self.errors = []
-        self.stats = Stat(self.new.files_count, self.old.files_count)
 
-        if self.new.path != '' and self.old.path != '':
+        try:
+            self.codebase1 = VirtualCodebase(new_path)
+            self.codebase2 = VirtualCodebase(old_path)
+            
+        except Exception as exception:
+            self.errors.append(str(exception))
+
+        if self.codebase1 is not None and self.codebase2 is not None:
```

Comment:
Can we just change this to 
```
if self.codebase1 is None or self.codebase2 is None:
    continue
rest of logic
```

This is easier to read and we can lose an indentation.

Reply:
Yes we can :)

Reply:
Changed in commit e6a6cbd